src/main/java/unicash/logic/commands/AddTransactionCommand.java

```diff
@@ -16,23 +23,28 @@
 public class AddTransactionCommand extends Command {
     public static final String COMMAND_WORD = "add_transaction";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a transaction to UniCa$h. \n"
-            + "\n"
-            + "Parameters: "
-            + CliSyntax.PREFIX_NAME + "NAME "
-            + CliSyntax.PREFIX_TYPE + "TYPE "
-            + CliSyntax.PREFIX_AMOUNT + "AMOUNT "
-            + CliSyntax.PREFIX_DATETIME + "DATETIME "
-            + CliSyntax.PREFIX_LOCATION + "LOCATION "
-            + "[" + CliSyntax.PREFIX_CATEGORY + "CATEGORY]...\n"
-            + "\n"
-            + "Example: " + COMMAND_WORD + " "
-            + CliSyntax.PREFIX_NAME + "Buying groceries "
-            + CliSyntax.PREFIX_TYPE + "expense "
-            + CliSyntax.PREFIX_AMOUNT + "300 "
-            + CliSyntax.PREFIX_DATETIME + "18-08-2023 19:30 "
-            + CliSyntax.PREFIX_LOCATION + "ntuc "
-            + CliSyntax.PREFIX_CATEGORY + "household";
+    public static final String MESSAGE_USAGE = new CommandUsage.Builder()
+            .setCommandWord(COMMAND_WORD)
+            .setDescription("Adds a transaction to UniCa$h.")
+            .setExample(
+                    ExampleGenerator.generate(
+                            "add_transaction",
+                            PREFIX_NAME,
+                            PREFIX_TYPE,
+                            PREFIX_AMOUNT,
+                            PREFIX_DATETIME,
+                            PREFIX_LOCATION,
+                            PREFIX_CATEGORY
+                    )
+            )
```

Comment:
I think we can refactor the generation of the example command to a separate method so I can reuse it to generate examples in the help command. But I can do it in my PR

Reply:
It is already a separate method call :o

Reply:
Oh i meant like within the command class itself. So a method like 
```
public String generateExample() {
return ExampleGenerator.generate(
                            "add_transaction",
                            PREFIX_NAME,
                            PREFIX_TYPE,
                            PREFIX_AMOUNT,
                            PREFIX_DATETIME,
                            PREFIX_LOCATION,
                            PREFIX_CATEGORY
                    )
```
for each of the commands. Else need to specify the prefix again

Reply:
Ahh, alright, that makes sense, I'll merge first and you can pick it up in your PR?

Reply:
Yeap sounds good